src/library_pthread.js

```diff
@@ -494,9 +495,9 @@ var LibraryPThread = {
   $cleanupThread: function(pthread_ptr) {
     if (ENVIRONMENT_IS_PTHREAD) throw 'Internal Error! cleanupThread() can only ever be called from main application thread!';
     if (!pthread_ptr) throw 'Internal Error! Null pthread_ptr in cleanupThread!';
-    {{{ makeSetValue('pthread_ptr', C_STRUCTS.pthread.self, 0, 'i32') }}};
     var pthread = PThread.pthreads[pthread_ptr];
```

Comment:
Can we add  comment here perhaps as to how it could be that `pthread_ptr` can sometimes find a valid thread in the `PThread.pthreads` and other times not?  (i.e. why is the if condition below needed).

Reply:
Added comment.